### ISSUE TITLE
Bug: (apparently) not ignoring dotted directories by default

### ISSUE THREAD

**Issue by alexrussell** [eslint/eslint#18236]
### Environment

Node version: v21.2.0
npm version: 10.2.3
Local ESLint version: v8.57.0
Global ESLint version: not installed globally
Operating System: macOS 13.6.2

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

Using the flat config style, file: `eslint.config.js` with `"type": "module"` in `package.json`.

```js
import eslint from '@eslint/js'

export default [
  eslint.configs.recommended,
]
```
</details>

<details>
<summary>`bad.js` file contents</summary>

```js
const a = 0
```
</details>

The contents of `bad.js` (see above) are to be placed in the following places, relative to the root of
the project:

- `lint/bad.js`
- `lint/not-ignored/bad.js`
- `lint/.ignored/bad.js`

Then simply run `npx eslint .`.

### What did you expect to happen?

I would expect that the only two files linted here would be the first two (i.e. `lint/bad.js` and `lint/not-ignored/bad.js`).

### What actually happened?

All three files are being included (and linted), see command output below:

<details>
<summary>Command output</summary>

```shell
$ npx eslint .

{cwd}/lint/.ignored/bad.js
  1:5  error  'a' is assigned a value but never used  no-unused-vars

{cwd}/lint/bad.js
  1:7  error  'a' is assigned a value but never used  no-unused-vars

{cwd}/lint/not-ignored/bad.js
  1:5  error  'a' is assigned a value but never used  no-unused-vars

✖ 3 problems (3 errors, 0 warnings)
```
</details>

If I specify an ignore pattern that ignores "dotted directories" then it correctly ignore that file. But I didn't expect to have to specify this. See command output below:

<details>
<summary>Command output</summary>

```shell
$ npx eslint --ignore-pattern '**/.*' .

{cwd}/lint/bad.js
  1:7  error  'a' is assigned a value but never used  no-unused-vars

{cwd}/lint/no-ignore/bad.js
  1:5  error  'a' is assigned a value but never used  no-unused-vars

✖ 2 problems (2 errors, 0 warnings)
```
</details>

Note that attempting this "please do actually ignore dotted directories" configuration inside an `.eslintignore` file doesn't seem to work, but it may be that I don't have the syntax quite right there, or maybe they don't work with flat configs and I haven't seen it in the docs.

### Link to Minimal Reproducible Example

https://github.com/alexrussell/eslint-ignore-demo

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

I can only assume I have done something wrong here, but I can't for the life of me see what it is! I do wonder if the semantics of eslint are quite different when using the new flat config, and maybe the docs aren't quite up-to-date with that?

**Comment by Tanujkanti4441** [comment]
Hi @alexrussell, thanks for the issue,

This works as intended. New flat-config use minimatch syntax to match files for linting which by default ignores directory or files starts with dot in their name unless `{dot:true}` is set, here we can see
 https://github.com/eslint/eslint/blob/9b7bd3be066ac1f72fa35c4d31a1b178c7e2b683/lib/eslint/eslint-helpers.js#L29

@mdjermanovic, can you please confirm the behavior.

i think we should document it.

**Comment by mdjermanovic** [comment]
Hi @alexrussell, thanks for the issue!

This works as intended. In flat config, dotfiles and dotfolders are not ignored by default.

 > Note that attempting this "please do actually ignore dotted directories" configuration inside an `.eslintignore` file doesn't seem to work, but it may be that I don't have the syntax quite right there, or maybe they don't work with flat configs and I haven't seen it in the docs.

The pattern is good, but flat config doesn't support `.eslintignore` files. You can specify ignore patterns directly in `eslint.config.js`:

https://eslint.org/docs/latest/use/configure/configuration-files-new#globally-ignoring-files-with-ignores

In your example, the configuration would be:

```js
// eslint.config.js

import eslint from '@eslint/js'

export default [
  {
    ignores: ['**/.*']
  },
  eslint.configs.recommended,
]

```
